orb-ui/cone/src/lcd.rs

```diff
@@ -40,33 +43,28 @@ pub enum LcdCommand {
 }
 
 impl Lcd {
-    pub(crate) fn spawn() -> eyre::Result<Lcd> {
-        let (tx, mut rx) = mpsc::unbounded_channel();
-        let (shutdown_signal, shutdown_receiver) = watch::channel(());
+    pub(crate) fn spawn() -> eyre::Result<(Lcd, LcdJoinHandle)> {
+        let (cmd_tx, mut cmd_rx) = mpsc::unbounded_channel();
+        let (kill_tx, kill_rx) = oneshot::channel();
 
         let task_handle =
-            task::spawn(
-                async move { handle_lcd_update(&mut rx, shutdown_receiver).await },
-            );
-
-        Ok(Lcd {
-            tx,
-            shutdown_signal,
-            task_handle: Some(task_handle),
-        })
+            task::spawn(async move { do_lcd_update(&mut cmd_rx, kill_rx).await });
+
+        Ok((Lcd { cmd_tx, kill_tx }, LcdJoinHandle(task_handle)))
     }
 
-    pub(crate) fn clone_tx(&self) -> mpsc::UnboundedSender<LcdCommand> {
-        self.tx.clone()
+    pub(crate) fn send(&mut self, cmd: LcdCommand) -> eyre::Result<()> {
```

Comment:
Just expose it as a ref. Gives a richer, more powerful api. Especially important if you switch to a bounded channel, as you'll need access to both send and try send, blocking send, etc.

Also, it should only require &self.